tests/application-operator-tests/test/testkit/wait.go

```diff
@@ -1,42 +1,44 @@
 package testkit
 
 import (
+	"errors"
 	"time"
 )
 
-func (ts *TestSuite) waitForFunction(conditionalFunc func() bool, message string, timeout time.Duration) {
+func WaitForFunction(interval, timeout time.Duration, conditionalFunc func() bool) error {
 	done := time.After(timeout)
 
 	for {
 		if conditionalFunc() {
-			return
+			return nil
 		}
 
 		select {
 		case <-done:
-			ts.t.Errorf(message)
+			return errors.New("timeout waiting for condition")
 		default:
-			time.Sleep(defaultCheckInterval)
+			time.Sleep(interval)
 		}
 	}
 }
 
-func (ts *TestSuite) shouldLastFor(conditionalFunc func() bool, message string, timeout time.Duration) {
+func ShouldLastFor(interval, timeout time.Duration, conditionalFunc func() bool) error {
 	done := time.After(timeout)
 
 	for {
 		if !conditionalFunc() {
-			ts.t.Errorf(message)
+			return errors.New("unexpected condition occurred when should not")
```

Comment:
This error message is a bit blurry, I would suggest using just the `unexpected condition occurred` - don't know whether it makes it any better though. Do you have any ideas?